mender-update/daemon/state_machine/state_machine.cpp

```diff
@@ -318,8 +321,19 @@ void StateMachine::LoadStateFromDb() {
 		return;
 	}
 
+	auto &store = ctx_.mender_context.GetMenderStoreDB();
+
 	if (!exp_loaded.value()) {
 		log::Debug("No existing deployment data, starting from initial state");
+
+		auto err = store.Remove(ctx_.mender_context.update_control_maps);
+		if (err != error::NoError) {
+			log::Error(
+				"Error removing " + ctx_.mender_context.update_control_maps
+				+ " key from database: " + err.String());
+			// Nothing we can do about it.
+		}
```

Comment:
This will forever log an Error when booting. I think is best to check if the key exists before removing it, and only then log an error.

Reply:
It's smart enough to [ignore removing a non-existing key](https://github.com/mendersoftware/mender/blob/feature-c%2B%2B-client/common/key_value_database/platform/lmdb/lmdb.cpp#L73-L76).